apps/web/src/components/Table/ResponsiveTable/ResponsiveTable.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import { useSearchParams } from "react-router-dom";
+import { useIntl } from "react-intl";
+import isEqual from "lodash/isEqual";
+import type { ColumnDef, RowSelectionState } from "@tanstack/react-table";
+import {
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  getPaginationRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+
+import Table from "./Table";
+import SearchForm from "./SearchForm";
+import ColumnDialog from "./ColumnDialog";
+import NullMessage, { NullMessageProps } from "./NullMessage";
+import RowSelection, { getRowSelectionColumn } from "./RowSelection";
+import useControlledTableState from "./useControlledTableState";
+import TablePagination from "./TablePagination";
+import { INITIAL_STATE, SEARCH_PARAM_KEY } from "./constants";
+
+import type {
+  AddLinkProps,
+  DatasetDownload,
+  DatasetPrint,
+  PaginationDef,
+  RowSelectDef,
+  SearchDef,
+  SortDef,
+} from "./types";
+
+interface TableProps<TData> {
+  /** Accessible name for the table */
+  caption: React.ReactNode;
+  /** Data to be displayed within the table */
+  data: TData[];
+  /** Column definitions for `react-table` */
+  columns: ColumnDef<TData>[];
+  /** Column definitions for `react-table` */
+  hiddenColumnIds: string[];
+  /** Determine if any aspect of the table is loading (server side) */
+  isLoading?: boolean;
+  /** Override default null message with a custom one */
+  nullMessage?: NullMessageProps;
+  /** Enable row selection */
+  rowSelect?: RowSelectDef<TData>;
+  /** Enable the search form */
+  search?: SearchDef<TData>;
+  /** Enable sorting */
+  sort?: SortDef;
+  /** Enable pagination */
+  pagination?: PaginationDef;
+  /** Enable printing selected rows (requires rowSelect) */
+  print?: DatasetPrint;
+  /** Enable downloading selected rows and/or all data (requires rowSelect) */
+  download?: DatasetDownload;
+  /** Enable the "add item" button */
+  add?: AddLinkProps;
+}
+
+const ResponsiveTable = <TData extends object>({
+  caption,
+  data,
+  columns,
+  hiddenColumnIds,
+  isLoading,
+  nullMessage,
+  rowSelect,
+  search,
+  sort,
+  download,
+  print,
+  add,
+  pagination,
+}: TableProps<TData>) => {
+  const id = React.useId();
+  const intl = useIntl();
+  const [, setSearchParams] = useSearchParams();
+  const isInternalSearch = search && search.internal;
+  const memoizedData = React.useMemo(() => data, [data]);
```

Comment:
Is using `useMemo` like this any different from using data directly?

Reply:
Huh, for some reason I thought the data needed to be memoized for `react-table` but I can't find anything to confirm that so maybe we can just pass it in directly! 🤔 I think they also removed the requirement for columns to memoized as well 🤷‍♀️